addons/website_sale/static/src/js/website_sale.js

```diff
@@ -589,6 +591,24 @@ sAnimations.registry.WebsiteSale = sAnimations.Class.extend(ProductConfiguratorM
     _onChangeShippingUseSame: function (ev) {
         $('.ship_to_other').toggle(!$(ev.currentTarget).prop('checked'));
     },
+    /*
+     * @override
+     *
+     * Dirty fix: prevent options modal events to be triggered and bubbled
+     */
+    onChangeVariant: function (ev) {
+        if (ev.originalEvent && Array.isArray(ev.originalEvent.path)) {
+            var nodes = ev.originalEvent.path;
+            for (var i=nodes.length-1; i >= 0; i--) {
+                var $node = $(nodes[i]);
+                if ($node.hasClass('oe_optional_products_modal')) {
```

Comment:
`$(ev.originalEvent.path).hasClass('oe_optional_products_modal')` does the same thing as the loop and stuff.

But I am not sure to understand the fix. Should  not it be `stopPropagation` in all cases ?

Reply:
Well, this is the whole problem.
2 handlers are bound to the event:
- the one on the ProductConfiguratorMixin, instanciated by the ProductConfiguratorModal
- the one on the widget WebsiteSale
Those two widgets are not even siblings, as the Modal hooks WebsiteSale (it is the same instance as the one created for the whole page -- ) through its CSS class .oe_website_sale.
So the fix in WebsiteSale just checks if the event comes from the modal, in which case the WebsiteSale widget shoud not react.
This is really an interpretation on my part, I'm not really sure which part of websiteSale's behavior the Modal should have....

Hence, stopPropagation maybe for all cases, but here, really the return has relevance

Reply:
If the problem is fixed and @seb-odoo is ok with the fix, ok for me once it is adapted to my comments :wink: